optuna/terminator/callback.py

```diff
@@ -0,0 +1,34 @@
+from typing import Optional
+
+from optuna.logging import get_logger
+from optuna.study.study import Study
+from optuna.terminator.improvement.evaluator import DEFAULT_MIN_N_TRIALS
+from optuna.terminator.terminator import BaseTerminator
+from optuna.terminator.terminator import Terminator
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+_logger = get_logger(__name__)
+
+
+class TerminatorCallback:
```

Comment:
Is it time to start considering `optuna.callbacks` ? We'll have `optuna.terminator.callback.TerminatorCallback` and `optuna.study.MaxTrialsCallback` with similar behaviors and APIs.

Reply:
Hmm, I think there are pros and cons of putting all callback classes in `optuna.callbacks`. The module structure may be simple for the user, but it also accumulates dependencies of the callback functionalities. In particular, the GP-related dependencies of `TerminatorCallback` may make the dependency puzzle even more complex.

Reply:
I agree with @g-votte.

In terms of simplicity of module structure, It would be clearer to users if the all terminator-related APIs (e.g. `TerminatorCallback`, `report_cross_validation_scores`, etc.) are placed under the `optuna.terminator` module.

Reply:
Sounds fair, at least for now, given that there also are multiple public components to the terminator besides the callback class itself. Thank you for the discussion.